Build failure: Compilation of the tcl-8_5 package fails under the x86-64 architecture.

## Steps To Reproduce

Steps to reproduce the behavior:

1. create nix-shell file
  ```nix
{ pkgs ? import <nixpkgs> { } }:

pkgs.mkShell {
  buildInputs = with pkgs; [
    clang
    cmake
    llvmPackages.libcxx
    xz.dev
    tcl-9_0
    tcl-8_5
    tk-9_0.dev
    sqlite.dev
    zlib.dev
    libffi.dev
    readline.dev
    libedit.dev
    bzip2.dev
    libxcrypt
    openssl.dev
    ncurses.dev
    pkg-config
  ];

  shellHook = ''
    export TCL_LIBRARY="${pkgs.tcl-9_0}/lib/tcl${pkgs.tcl.version}"
    export TK_LIBRARY="${pkgs.tk-9_0}/lib/tk${pkgs.tk.version}"
  '';
}
 ```
2. nix-shell the file

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/5ff2i3msiar0mmfc22nvm5n190iqcd7g-tcl8.5.19-src.tar.gz
source root is tcl8.5.19
setting SOURCE_DATE_EPOCH to timestamp 1454964293 of file tcl8.5.19/win/tclsh.exe.manifest.in
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
substituteStream() in derivation tcl-8.5.19: WARNING: '--replace' is deprecated, use --replace-{fail,war>
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc028>
configure flags: --prefix=/nix/store/27zin8zqx4qk01p1f7i0cq8d3jp80666-tcl-8.5.19 --enable-threads --mand>
checking whether to use symlinks for manpages... yes
checking whether to compress the manpages... no
checking whether to add a package name suffix for the manpages... no
checking for gcc... gcc
checking for C compiler default output file name... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables...
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ANSI C... none needed
checking for inline... inline
checking how to run the C preprocessor... gcc -E
checking for egrep... grep -E
checking for ANSI C header files... no
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking dirent.h... yes
checking float.h usability... yes
checking float.h presence... yes
checking for float.h... yes
checking values.h usability... yes
checking values.h presence... yes
checking for values.h... yes
checking limits.h usability... yes
checking limits.h presence... yes
checking for limits.h... yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking sys/wait.h usability... yes
checking sys/wait.h presence... yes
checking for sys/wait.h... yes
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking sys/param.h usability... yes
checking sys/param.h presence... yes
checking for sys/param.h... yes
checking if the compiler understands -pipe... yes
checking for pthread_mutex_init in -lpthread... yes
checking for pthread_attr_setstacksize... yes
checking for pthread_atfork... yes
checking for pthread_attr_get_np... no
checking for pthread_getattr_np... yes
checking for pthread_getattr_np declaration... missing
checking for building with threads... yes
checking for sin... no
checking for main in -lieee... no
checking for main in -linet... no
checking net/errno.h usability... no
checking net/errno.h presence... no
checking for net/errno.h... no
checking for connect... yes
checking for gethostbyname... yes
checking how to build libraries... shared
checking for ranlib... ranlib
checking if 64bit support is requested... yes
checking if 64bit Sparc VIS support is requested... no
checking if compiler supports visibility "hidden"... yes
checking if rpath support is requested... yes
checking system version... Linux-6.6.67
checking for dlopen in -ldl... yes
checking for ar... ar
checking if compiler accepts -m64 flag... yes
checking for cast to union support... yes
checking for build with symbols... no
checking for required early compiler flags...  _LARGEFILE64_SOURCE
checking for 64-bit integer type... using long
checking whether byte ordering is bigendian... no
checking for getcwd... yes
checking for opendir... yes
checking for strtol... yes
checking for waitpid... yes
checking for strerror... yes
checking for getwd... yes
checking for wait3... yes
checking for uname... yes
checking for realpath... yes
checking for getaddrinfo... yes
checking for working getaddrinfo... yes
checking for getpwuid_r... yes
checking for getpwuid_r with 5 args... yes
checking for getpwnam_r... yes
checking for getpwnam_r with 5 args... yes
checking for getgrgid_r... yes
checking for getgrgid_r with 5 args... yes
checking for getgrnam_r... yes
checking for getgrnam_r with 5 args... yes
checking for gethostbyname_r... yes
checking for gethostbyname_r with 6 args... yes
checking for gethostbyaddr_r... yes
checking for gethostbyaddr_r with 7 args... no
checking for gethostbyaddr_r with 8 args... yes
checking sys/modem.h usability... no
checking sys/modem.h presence... no
checking for sys/modem.h... no
checking termios vs. termio vs. sgtty... termios
checking for fd_set in sys/types... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking whether time.h and sys/time.h may both be included... yes
checking for gmtime_r... yes
checking for localtime_r... yes
checking for mktime... yes
checking tm_tzadj in struct tm... no
checking tm_gmtoff in struct tm... yes
checking long timezone variable... no
checking time_t timezone variable... no
checking for struct stat.st_blocks... yes
checking for struct stat.st_blksize... yes
checking for blkcnt_t... yes
checking for fstatfs... yes
checking for working memcmp... yes
checking for memmove... yes
checking for strstr... yes
checking proper strstr implementation... broken
checking for strtoul... yes
checking proper strtoul implementation... broken
checking for strtod... yes
checking proper strtod implementation... (cached) ok
checking for strtod... (cached) yes
checking for Solaris2.4/Tru64 strtod bugs... buggy
checking for mode_t... yes
checking for pid_t... yes
checking for size_t... yes
checking for uid_t in sys/types.h... yes
checking for socklen_t... yes
checking for intptr_t... yes
checking for uintptr_t... yes
checking for opendir... (cached) yes
checking union wait... no
checking for strncasecmp... yes
checking for gettimeofday... yes
checking for gettimeofday declaration... present
checking whether char is unsigned... no
checking signed char declarations... yes
checking for a putenv() that copies the buffer... yes
checking langinfo.h usability... yes
checking langinfo.h presence... yes
checking for langinfo.h... yes
checking whether to use nl_langinfo... yes
checking for chflags... no
checking isnan... yes
checking for fts... yes
checking sys/ioctl.h usability... yes
checking sys/ioctl.h presence... yes
checking for sys/ioctl.h... yes
checking sys/filio.h usability... no
checking sys/filio.h presence... no
checking for sys/filio.h... no
checking system version... (cached) Linux-6.6.67
checking FIONBIO vs. O_NONBLOCK for nonblocking I/O... O_NONBLOCK
checking whether to use dll unloading... yes
checking for timezone data... supplied by OS vendor
checking whether to enable DTrace support... no
checking if the C stack grows upwards in memory... no
checking whether the cpuid instruction is usable... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating dltest/Makefile
config.status: creating tclConfig.sh
config.status: creating tcl.pc
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j16 SHELL=/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
/build/tcl8.5.19/unix/../generic/tclClock.c: In function 'ClockGetdatefieldsObjCmd':
/build/tcl8.5.19/unix/../generic/tclClock.c:1076:28: warning: '__builtin___sprintf_chk' may write a term>
 1076 |     sprintf(buffer+1, "%02d", diff / 3600);
      |                            ^
In file included from /nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/stdio.h:970,
                 from /build/tcl8.5.19/unix/../generic/tcl.h:139,
                 from /build/tcl8.5.19/unix/../generic/tclPort.h:25,
                 from /build/tcl8.5.19/unix/../generic/tclInt.h:36,
                 from /build/tcl8.5.19/unix/../generic/tclClock.c:16:
In function 'sprintf',
    inlined from 'ConvertUTCToLocalUsingC' at /build/tcl8.5.19/unix/../generic/tclClock.c:1076:5,
    inlined from 'ConvertUTCToLocal' at /build/tcl8.5.19/unix/../generic/tclClock.c:938:9,
    inlined from 'ClockGetdatefieldsObjCmd' at /build/tcl8.5.19/unix/../generic/tclClock.c:450:9:
/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/bits/stdio2.h:30:10: note: '__buil>
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
In file included from /build/tcl8.5.19/unix/../generic/tclExecute.c:17:
/build/tcl8.5.19/unix/../generic/tclExecute.c: In function 'VarHashCreateVar':
/build/tcl8.5.19/unix/../generic/tclInt.h:3910:33: warning: cast from pointer to integer of different si>
 3910 | #define TclOffset(type, field) ((int) ((char *) &((type *) 0)->field))
      |                                 ^
/build/tcl8.5.19/unix/../generic/tclExecute.c:165:30: note: in expansion of macro 'TclOffset'
  165 |     ((Var *) ((char *)hPtr - TclOffset(VarInHash, entry)))
      |                              ^~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclExecute.c:179:12: note: in expansion of macro 'VarHashGetValue'
  179 |     return VarHashGetValue(hPtr);
      |            ^~~~~~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclExecute.c: In function 'TclExecuteByteCode':
/build/tcl8.5.19/unix/../generic/tclExecute.c:2346:23: warning: cast from pointer to integer of differen>
 2346 |                     - (ptrdiff_t) objPtr->internalRep.twoPtrValue.ptr1;
      |                       ^
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
/build/tcl8.5.19/unix/../generic/tclExecute.c:7427:17: warning: cast from pointer to integer of differen>
 7427 |                 (ptrdiff_t) expandNestList->internalRep.twoPtrValue.ptr1))) {
      |                 ^
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
        -DCFG_INSTALL_LIBDIR="\"/nix/store/27zin8zqx4qk01p1f7i0cq8d3jp80666-tcl-8.5.19/lib\"" \
        -DCFG_INSTALL_BINDIR="\"/nix/store/27zin8zqx4qk01p1f7i0cq8d3jp80666-tcl-8.5.19/bin\"" \
        -DCFG_INSTALL_SCRDIR="\"/nix/store/27zin8zqx4qk01p1f7i0cq8d3jp80666-tcl-8.5.19/lib/tcl8.5\"" \
        -DCFG_INSTALL_INCDIR="\"/nix/store/27zin8zqx4qk01p1f7i0cq8d3jp80666-tcl-8.5.19/include\"" \
        -DCFG_INSTALL_DOCDIR="\"/nix/store/27zin8zqx4qk01p1f7i0cq8d3jp80666-tcl-8.5.19/share/man\"" \
        \
        -DCFG_RUNTIME_LIBDIR="\"/nix/store/27zin8zqx4qk01p1f7i0cq8d3jp80666-tcl-8.5.19/lib\"" \
        -DCFG_RUNTIME_BINDIR="\"/nix/store/27zin8zqx4qk01p1f7i0cq8d3jp80666-tcl-8.5.19/bin\"" \
        -DCFG_RUNTIME_SCRDIR="\"/nix/store/27zin8zqx4qk01p1f7i0cq8d3jp80666-tcl-8.5.19/lib/tcl8.5\"" \
        -DCFG_RUNTIME_INCDIR="\"/nix/store/27zin8zqx4qk01p1f7i0cq8d3jp80666-tcl-8.5.19/include\"" \
        -DCFG_RUNTIME_DOCDIR="\"/nix/store/27zin8zqx4qk01p1f7i0cq8d3jp80666-tcl-8.5.19/share/man\"" \
        \
        /build/tcl8.5.19/unix/../generic/tclPkgConfig.c
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
/build/tcl8.5.19/unix/../generic/tclObj.c: In function 'ParseBoolean':
/build/tcl8.5.19/unix/../generic/tclObj.c:2037:23: warning: writing 1 byte into a region of size 0 [-Wst>
 2037 |     lowerCase[length] = 0;
      |     ~~~~~~~~~~~~~~~~~~^~~
/build/tcl8.5.19/unix/../generic/tclObj.c:1995:10: note: at offset [-2147483648, -1] into destination ob>
 1995 |     char lowerCase[6], *str = TclGetStringFromObj(objPtr, &length);
      |          ^~~~~~~~~
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
In file included from /build/tcl8.5.19/unix/../generic/tclPort.h:25,
                 from /build/tcl8.5.19/unix/../generic/tclInt.h:36,
                 from /build/tcl8.5.19/unix/../generic/tclStringObj.c:37:
/build/tcl8.5.19/unix/../generic/tclStringObj.c: In function 'GrowUnicodeBuffer':
/build/tcl8.5.19/unix/../generic/tclStringObj.c:114:21: warning: '*' in boolean context, suggest '&&' in>
  114 |         ((numChars) * sizeof(Tcl_UniChar))
      |                     ^
/build/tcl8.5.19/unix/../generic/tcl.h:2292:55: note: in definition of macro 'attemptckrealloc'
 2292 | #   define attemptckrealloc(x,y) Tcl_AttemptRealloc(x,y)
      |                                                       ^
/build/tcl8.5.19/unix/../generic/tclStringObj.c:129:28: note: in expansion of macro 'STRING_SIZE'
  129 |                 (unsigned) STRING_SIZE(STRING_UALLOC(numChars)) )
      |                            ^~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclStringObj.c:129:40: note: in expansion of macro 'STRING_UALLOC'
  129 |                 (unsigned) STRING_SIZE(STRING_UALLOC(numChars)) )
      |                                        ^~~~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclStringObj.c:190:19: note: in expansion of macro 'stringAttemptReallo>
  190 |             ptr = stringAttemptRealloc(stringPtr, attempt);
      |                   ^~~~~~~~~~~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclStringObj.c:114:21: warning: '*' in boolean context, suggest '&&' in>
  114 |         ((numChars) * sizeof(Tcl_UniChar))
      |                     ^
/build/tcl8.5.19/unix/../generic/tcl.h:2292:55: note: in definition of macro 'attemptckrealloc'
 2292 | #   define attemptckrealloc(x,y) Tcl_AttemptRealloc(x,y)
      |                                                       ^
/build/tcl8.5.19/unix/../generic/tclStringObj.c:129:28: note: in expansion of macro 'STRING_SIZE'
  129 |                 (unsigned) STRING_SIZE(STRING_UALLOC(numChars)) )
      |                            ^~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclStringObj.c:129:40: note: in expansion of macro 'STRING_UALLOC'
  129 |                 (unsigned) STRING_SIZE(STRING_UALLOC(numChars)) )
      |                                        ^~~~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclStringObj.c:202:19: note: in expansion of macro 'stringAttemptReallo>
  202 |             ptr = stringAttemptRealloc(stringPtr, attempt);
      |                   ^~~~~~~~~~~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclStringObj.c:114:21: warning: '*' in boolean context, suggest '&&' in>
  114 |         ((numChars) * sizeof(Tcl_UniChar))
      |                     ^
/build/tcl8.5.19/unix/../generic/tcl.h:2290:41: note: in definition of macro 'ckrealloc'
 2290 | #   define ckrealloc(x,y) Tcl_Realloc(x,y)
      |                                         ^
/build/tcl8.5.19/unix/../generic/tclStringObj.c:126:28: note: in expansion of macro 'STRING_SIZE'
  126 |                 (unsigned) STRING_SIZE(STRING_UALLOC(numChars)) )
      |                            ^~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclStringObj.c:126:40: note: in expansion of macro 'STRING_UALLOC'
  126 |                 (unsigned) STRING_SIZE(STRING_UALLOC(numChars)) )
      |                                        ^~~~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclStringObj.c:208:15: note: in expansion of macro 'stringRealloc'
  208 |         ptr = stringRealloc(stringPtr, attempt);
      |               ^~~~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclStringObj.c: In function 'Tcl_SetObjLength':
/build/tcl8.5.19/unix/../generic/tclStringObj.c:114:21: warning: '*' in boolean context, suggest '&&' in>
  114 |         ((numChars) * sizeof(Tcl_UniChar))
      |                     ^
/build/tcl8.5.19/unix/../generic/tcl.h:2290:41: note: in definition of macro 'ckrealloc'
 2290 | #   define ckrealloc(x,y) Tcl_Realloc(x,y)
      |                                         ^
/build/tcl8.5.19/unix/../generic/tclStringObj.c:126:28: note: in expansion of macro 'STRING_SIZE'
  126 |                 (unsigned) STRING_SIZE(STRING_UALLOC(numChars)) )
      |                            ^~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclStringObj.c:126:40: note: in expansion of macro 'STRING_UALLOC'
  126 |                 (unsigned) STRING_SIZE(STRING_UALLOC(numChars)) )
      |                                        ^~~~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclStringObj.c:856:25: note: in expansion of macro 'stringRealloc'
  856 |             stringPtr = stringRealloc(stringPtr, length);
      |                         ^~~~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclStringObj.c: In function 'Tcl_AttemptSetObjLength':
/build/tcl8.5.19/unix/../generic/tclStringObj.c:114:21: warning: '*' in boolean context, suggest '&&' in>
  114 |         ((numChars) * sizeof(Tcl_UniChar))
      |                     ^
/build/tcl8.5.19/unix/../generic/tcl.h:2292:55: note: in definition of macro 'attemptckrealloc'
 2292 | #   define attemptckrealloc(x,y) Tcl_AttemptRealloc(x,y)
      |                                                       ^
/build/tcl8.5.19/unix/../generic/tclStringObj.c:129:28: note: in expansion of macro 'STRING_SIZE'
  129 |                 (unsigned) STRING_SIZE(STRING_UALLOC(numChars)) )
      |                            ^~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclStringObj.c:129:40: note: in expansion of macro 'STRING_UALLOC'
  129 |                 (unsigned) STRING_SIZE(STRING_UALLOC(numChars)) )
      |                                        ^~~~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclStringObj.c:986:25: note: in expansion of macro 'stringAttemptReallo>
  986 |             stringPtr = stringAttemptRealloc(stringPtr, length);
      |                         ^~~~~~~~~~~~~~~~~~~~
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
In file included from /build/tcl8.5.19/unix/../generic/tclVar.c:20:
/build/tcl8.5.19/unix/../generic/tclVar.c: In function 'VarHashCreateVar':
/build/tcl8.5.19/unix/../generic/tclInt.h:3910:33: warning: cast from pointer to integer of different si>
 3910 | #define TclOffset(type, field) ((int) ((char *) &((type *) 0)->field))
      |                                 ^
/build/tcl8.5.19/unix/../generic/tclVar.c:48:30: note: in expansion of macro 'TclOffset'
   48 |     ((Var *) ((char *)hPtr - TclOffset(VarInHash, entry)))
      |                              ^~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclVar.c:67:16: note: in expansion of macro 'VarHashGetValue'
   67 |         return VarHashGetValue(hPtr);
      |                ^~~~~~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclVar.c: In function 'VarHashFirstVar':
/build/tcl8.5.19/unix/../generic/tclInt.h:3910:33: warning: cast from pointer to integer of different si>
 3910 | #define TclOffset(type, field) ((int) ((char *) &((type *) 0)->field))
      |                                 ^
/build/tcl8.5.19/unix/../generic/tclVar.c:48:30: note: in expansion of macro 'TclOffset'
   48 |     ((Var *) ((char *)hPtr - TclOffset(VarInHash, entry)))
      |                              ^~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclVar.c:94:16: note: in expansion of macro 'VarHashGetValue'
   94 |         return VarHashGetValue(hPtr);
      |                ^~~~~~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclVar.c: In function 'VarHashNextVar':
/build/tcl8.5.19/unix/../generic/tclInt.h:3910:33: warning: cast from pointer to integer of different si>
 3910 | #define TclOffset(type, field) ((int) ((char *) &((type *) 0)->field))
      |                                 ^
/build/tcl8.5.19/unix/../generic/tclVar.c:48:30: note: in expansion of macro 'TclOffset'
   48 |     ((Var *) ((char *)hPtr - TclOffset(VarInHash, entry)))
      |                              ^~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclVar.c:107:16: note: in expansion of macro 'VarHashGetValue'
  107 |         return VarHashGetValue(hPtr);
      |                ^~~~~~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclVar.c: In function 'Tcl_ArrayObjCmd':
/build/tcl8.5.19/unix/../generic/tclInt.h:3910:33: warning: cast from pointer to integer of different si>
 3910 | #define TclOffset(type, field) ((int) ((char *) &((type *) 0)->field))
      |                                 ^
/build/tcl8.5.19/unix/../generic/tclVar.c:48:30: note: in expansion of macro 'TclOffset'
   48 |     ((Var *) ((char *)hPtr - TclOffset(VarInHash, entry)))
      |                              ^~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclVar.c:2881:32: note: in expansion of macro 'VarHashGetValue'
 2881 |                 Var *varPtr2 = VarHashGetValue(searchPtr->nextEntry);
      |                                ^~~~~~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclInt.h:3910:33: warning: cast from pointer to integer of different si>
 3910 | #define TclOffset(type, field) ((int) ((char *) &((type *) 0)->field))
      |                                 ^
/build/tcl8.5.19/unix/../generic/tclVar.c:48:30: note: in expansion of macro 'TclOffset'
   48 |     ((Var *) ((char *)hPtr - TclOffset(VarInHash, entry)))
      |                              ^~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclVar.c:2954:23: note: in expansion of macro 'VarHashGetValue'
 2954 |             varPtr2 = VarHashGetValue(hPtr);
      |                       ^~~~~~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclInt.h:3910:33: warning: cast from pointer to integer of different si>
 3910 | #define TclOffset(type, field) ((int) ((char *) &((type *) 0)->field))
      |                                 ^
/build/tcl8.5.19/unix/../generic/tclVar.c:48:30: note: in expansion of macro 'TclOffset'
   48 |     ((Var *) ((char *)hPtr - TclOffset(VarInHash, entry)))
      |                              ^~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclVar.c:3278:39: note: in expansion of macro 'VarHashGetValue'
 3278 |                     protectedVarPtr = VarHashGetValue(search.nextEntryPtr);
      |                                       ^~~~~~~~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclVar.c: In function 'FreeVarEntry':
/build/tcl8.5.19/unix/../generic/tclInt.h:3910:33: warning: cast from pointer to integer of different si>
 3910 | #define TclOffset(type, field) ((int) ((char *) &((type *) 0)->field))
      |                                 ^
/build/tcl8.5.19/unix/../generic/tclVar.c:48:30: note: in expansion of macro 'TclOffset'
   48 |     ((Var *) ((char *)hPtr - TclOffset(VarInHash, entry)))
      |                              ^~~~~~~~~
/build/tcl8.5.19/unix/../generic/tclVar.c:5635:19: note: in expansion of macro 'VarHashGetValue'
 5635 |     Var *varPtr = VarHashGetValue(hPtr);
      |                   ^~~~~~~~~~~~~~~
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
        -DTCL_PACKAGE_PATH="\"/nix/store/27zin8zqx4qk01p1f7i0cq8d3jp80666-tcl-8.5.19/lib \"" \
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
        -DTCL_PACKAGE_PATH="\"/nix/store/27zin8zqx4qk01p1f7i0cq8d3jp80666-tcl-8.5.19/lib \"" \
        /build/tcl8.5.19/unix/../unix/tclUnixInit.c
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
gcc -c -O2  -pipe -m64  -Wall -fPIC -I"." -I/build/tcl8.5.19/unix/../unix -I/build/tcl8.5.19/unix/../gen>
In file included from /build/tcl8.5.19/unix/../unix/tclUnixInit.c:12:
/nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/lib/gcc/x86_64-unknown-linux-gnu/14.2.1/incl>
  145 | typedef __PTRDIFF_TYPE__ ptrdiff_t;
      |                          ^~~~~~~~~
In file included from /build/tcl8.5.19/unix/../unix/tclUnixInit.c:11:
/build/tcl8.5.19/unix/../generic/tclInt.h:54:13: note: previous declaration of 'ptrdiff_t' with type 'pt>
   54 | typedef int ptrdiff_t;
      |             ^~~~~~~~~
make: *** [Makefile:1464: tclUnixInit.o] Error 1
make: *** Waiting for unfinished jobs....
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.67, NixOS, 25.05 (Warbler), 25.05.20241227.634fd46`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

@agbrooks

--- a/pkgs/development/interpreters/tcl/generic.nix
+++ b/pkgs/development/interpreters/tcl/generic.nix
@@ -62,6 +62,10 @@ let
         # Don't install tzdata because NixOS already has a more up-to-date copy.
         "--with-tzdata=no"
       ]
+      ++ lib.optionals (lib.versionOlder version "8.6") [
+        # configure check broke due to GCC 14
+        "ac_cv_header_stdc=yes"
+      ]
       ++ lib.optionals (lib.versionAtLeast version "9.0") [
         # By default, tcl libraries get zipped and embedded into libtcl*.so,
         # which gets `zipfs mount`ed at runtime. This is fragile (for example

